alf/environments/metadrive/geometry.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) 2022 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+from typing import NamedTuple, Optional
+
+import numpy as np
+
+try:
+    import metadrive
+    from metadrive.constants import LineType
+    from metadrive.component.lane.metadrive_lane import MetaDriveLane
+except ImportError:
+    from unittest.mock import Mock
+    # create 'metadrive' as a mock to not break python argument type hints
+    metadrive = Mock()
+
+
+class CategoryEncoder(object):
+    """A category encoder can 
+       
+    1. Converts integer categories into their corresponding one-hot encoding.
+    2. Translate the type of driving scenario objects into its integer category.
+
+    """
+
+    def __init__(self):
+        self._num_types = 4
```

Comment:
I'm thinking of whether there might be a requirement for changing this number in the future. If yes, it might be better to move this as an argument to the constructor?

Reply:
Yes it is possible we have such needs in the future, but in that case I think we can just update the `4` here. Reason being that

1. The documentation cost and readability is better.
2. If we have such need, it is more likely applied uniformly